src/types.ts

```diff
@@ -46,10 +48,14 @@ export type RequestPayloadOf<
   Route extends keyof Endpoints,
 > = Endpoints[Route]['Request'] & PathParamsOf<Route>;
 
-type RestrictedUseQueryOptions<Response> = Omit<
-  UseQueryOptions<Response, unknown>,
-  'queryKey' | 'queryFn'
-> & {
+type QueryOptions<Response> =
+  | UseQueryOptions<Response, unknown>
+  | UseInfiniteQueryOptions<Response, unknown>;
+
+type RestrictedUseQueryOptions<
+  Response,
+  TQueryOptions extends QueryOptions<Response> = UseQueryOptions<Response>,
+> = Omit<TQueryOptions, 'queryKey' | 'queryFn'> & {
```

Comment:
I think these types are a bit hard to read + understand (even though they are correct). I think it might be more obvious to future devs if we just made a a separate type for `RestrictedUseInfiniteQueryOptions` (even though it might result in a bit of duplication). Thoughts?

Reply:
If we do keep the smarter types, I think it's worth leaving some comments about them.